cmd/state-svc/internal/resolver/analytics/analytics.go

```diff
@@ -225,36 +228,64 @@ func (r *Resolver) eventLoop() {
 	tick := time.NewTicker(time.Minute * 5)
 	defer tick.Stop()
 
+	projectIDMap := make(map[string]string)
+
 	// flush the deferred data initially
-	if err := r.flushDeferred(); err != nil {
+	if err := r.flushDeferred(projectIDMap); err != nil {
 		logging.Error("Failed to flush deferred data: %s", errs.JoinMessage(err))
 	}
 	for {
 		select {
 		case <-r.ctx.Done():
 			return
 		case <-tick.C:
-			if err := r.flushDeferred(); err != nil {
+			if err := r.flushDeferred(projectIDMap); err != nil {
 				logging.Error("Failed to flush deferred data: %s", errs.JoinMessage(err))
 			}
 		case ev := <-r.events:
-			r.event(ev)
+			r.event(ev, projectIDMap)
 		}
 	}
 }
 
-func (r *Resolver) flushDeferred() error {
+func (r *Resolver) flushDeferred(projectIDMap map[string]string) error {
 	events, err := deferred.LoadEvents()
 	if err != nil {
 		return errs.Wrap(err, "Failed to load deferred events")
 	}
 	for _, event := range events {
-		r.event(event)
+		r.event(event, projectIDMap)
 	}
 
 	return nil
 }
 
+// projectID resolves the projectID from projectName and caches the result in the provided projectIDMap
+func projectID(projectIDMap map[string]string, projectName string) string {
```

Comment:
I suggest instead relying on `github.com/patrickmn/go-cache`, which we're already using elsewhere on the state-svc.